### ISSUE TITLE
Build failure with gcc/g++ 13: undefined reference to `WasmEdge::Fault::emitFault(WasmEdge::ErrCode)'

### ISSUE THREAD

**Issue by paravoid** [WasmEdge/WasmEdge#2695]
## Description

Side-note/context: WasmEdge is now in Debian unstable & testing (codename trixie). There is already a reverse dependency, as I've enabled WasmEdge support in crun.

During a regular Debian archive rebuild, WasmEdge failed to (re)build from source (FTBFS). This was reported as [Debian Bug #1042193](https://bugs.debian.org/1042193).

I've tracked it down to the switch from GCC 12 -> GCC-13, i.e. rebuilding with CC=gcc-12 CXX=g++-12 works around the issue

The full build log as linked from the bug, is: http://qa-logs.debian.net/2023/07/26/wasmedge_0.13.1+dfsg-1_unstable.log

Relevant part:
```
> /usr/bin/c++ -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -Wl,-z,relro -Wl,-z,now -Wl,--export-dynamic -rdynamic CMakeFiles/wasmedge.dir/wasmedge.cpp.o -o wasmedge  -Wl,-rpath,"\$ORIGIN/../../lib/api::::::::::" ../../lib/api/libwasmedge.so.0.0.3 /usr/lib/x86_64-linux-gnu/libspdlog.so.1.10.0 /usr/lib/x86_64-linux-gnu/libfmt.so.9.1.0 
> /usr/bin/ld: ../../lib/api/libwasmedge.so.0.0.3: undefined reference to `WasmEdge::Fault::emitFault(WasmEdge::ErrCode)'
> collect2: error: ld returned 1 exit status
```

I think what's happening is that the method in `lib/system/fault.cpp` is inlined and optimized away, and is not available to its callsite, `lib/executor/engine/proxy.cpp`. I am not sure, but I tried this and it worked:
```diff
--- a/lib/system/fault.cpp
+++ b/lib/system/fault.cpp
@@ -114,7 +114,7 @@ Fault::~Fault() noexcept {
   localHandler = std::exchange(Prev, nullptr);
 }
 
-[[noreturn]] inline void Fault::emitFault(ErrCode Error) {
+[[noreturn]] void Fault::emitFault(ErrCode Error) {
   assuming(localHandler != nullptr);
   longjmp(localHandler->Buffer, static_cast<int>(Error.operator uint32_t()));
 }
```

## Environment

- Hardware Architecture: x86\_64
- Operating system: Debian unstable
- C++ Compiler version: g++ (Debian 13.1.0-9) 13.1.0
- CMake version: 3.27.1
- CMake flags: -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=None -DCMAKE_INSTALL_SYSCONFDIR=/etc -DCMAKE_INSTALL_LOCALSTATEDIR=/var -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DFETCHCONTENT_FULLY_DISCONNECTED=ON -DCMAKE_INSTALL_RUNSTATEDIR=/run -DCMAKE_SKIP_INSTALL_ALL_DEPENDENCY=ON "-GUnix Makefiles" -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_INSTALL_LIBDIR=lib/x86_64-linux-gnu -DWASMEDGE_BUILD_PLUGINS=OFF
- Boost version: 1.74.0

## Steps to Reproduce

podman run -it --rm debian:sid
sed -i 's/deb$/deb deb-src/' /etc/apt/sources.list.d/debian.sources
apt update && apt dist-upgrade
apt build-dep wasmedge
apt source wasmedge
cd wasmedge*
dpkg-buildpackage -b -uc -us

**Comment by hydai** [comment]
Hi @paravoid 
Thanks for this. I am going to check it. Will get back to you next Monday.

One thing is that `boost` dependency is no longer needed in the 0.13.x series. So you don't need to install it anymore.

**Comment by hydai** [comment]
Hi @paravoid 
I can reproduce this. And the current workaround is just like your solution. So please use the patch for the workaround. We will fix this in the next release, 0.14.0, planned for the end of Aug.

**Comment by paravoid** [comment]
Thanks so much! This is patched locally, and uploaded with the 0.13.3 release in Debian.

**Comment by hydai** [comment]
@paravoid 
Would you like to have a tiny PR for this fix? So we can merge it and close this issue.
